ethereum_history_api/circuits/lib/src/misc/bytes.nr

```diff
@@ -1,27 +1,25 @@
-pub fn right_pad<N>(arr: [u8; N]) -> ([u8; N], u64) {
-    byte_value(arr)
+pub fn byte_to_nibbles(byte: u8) -> (u8, u8) {
+    let upper = byte / 16;
+    let lower = byte - 16 * upper;
+    assert(upper < 16, "Nibble value must be less than 16");
+    assert(lower < 16, "Nibble value must be less than 16");
```

Comment:
No need for asserts above - defensive programming distrusting other functions (inputs and outputs), not yourself. 
If in doubt about own functionality - write a test.